### PR TITLE
Add deprecation warnings to the getitem join syntax with :

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/matrixtable.py [hunk 1]
```diff
@@ -507,10 +508,21 @@ def __getitem__(self, item):
             if row_key is not None and col_key is not None:
                 return self.index_entries(row_key, col_key)
             elif row_key is not None and col_key is None:
+                warnings.warn('The mt[<row keys>, :] syntax is deprecated, and will be removed before 0.2 release.\n'
+                              '  Use one of the following instead:\n'
+                              '    mt.rows()[<row keys>]\n'
```
Comment (jigold) [inline python/hail/matrixtable.py]: Do we use this syntax anywhere in the tests, examples, or tutorials? Might be worth removing them now.

File: python/hail/matrixtable.py [hunk 2]
```diff
@@ -507,10 +508,21 @@ def __getitem__(self, item):
...
             elif row_key is None and col_key is not None:
+                warnings.warn('The mt[:, <col keys>] syntax is deprecated, and will be removed before 0.2 release.\n'
+                              '  Use one of the following instead:\n'
+                              '    mt.cols()[<col keys>]\n'
+                              '    mt.index_cols(<col keys>)', stacklevel=2)
                 return self.index_cols(*col_key)
             else:
+                warnings.warn('The mt[:, :] syntax is deprecated, and will be removed before 0.2 release.\n'
+                              '  Use the following instead:\n'
+                              '    mt.index_globals()', stacklevel=2)
```
Comment (jigold) [inline python/hail/matrixtable.py]: Can you add documentation for `index_*` if we're going to be pointing users to these methods?

Comment (tpoterba) [reply python/hail/matrixtable.py]: I'm going to do this next, but I don't see a reason to combine into one PR

File: python/hail/matrixtable.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/matrixtable.py]: Can I do this in a followup PR? I want to keep everything as small as possible

File: python/hail/matrixtable.py (hunk 2 above)
Comment (jigold) [reply python/hail/matrixtable.py]: I don't care if it's the same PR. But I don't think we should have warning messages before the documentation for the methods is in.

File: python/hail/matrixtable.py (hunk 1 above)
Comment (jigold) [reply python/hail/matrixtable.py]: Sure.

File: python/hail/matrixtable.py (hunk 2 above)
Comment (tpoterba) [reply python/hail/matrixtable.py]: ok, fair. Let's hold this PR until the followup is merged.

**Pull request merged**